ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -1045,6 +1046,24 @@ private void registerCommandTopic() {
 
     final String commandTopic = commandStore.getCommandTopicName();
 
+    // migration code
+    if (checkMigrationConditions(commandTopic)) {
+      log.warn("Migrating command topic from the service context Kafka to the command "
+          + "producer/consumer Kafka.");
+      KsqlInternalTopicUtils.ensureTopic(
+          commandTopic,
+          ksqlConfigNoPort,
+          internalTopicClient
+      );
+
+      try {
+        CommandTopicMigrationUtil.commandTopicMigration(commandTopic, restConfig, ksqlConfigNoPort);
+      } catch (final Exception e) {
+        internalTopicClient.deleteTopics(Collections.singletonList(commandTopic));
```

Comment:
if you run the migration in a transaction, then you don't need to delete the topic here